src/st/np3/player_water_effect.c

```diff
@@ -0,0 +1,343 @@
+#include "np3.h"
+
+static u16 func_801C4144(s16 arg0, s16 arg1, s16 arg2, s16* arg3) {
+    s16 temp;
+    s16* ptr;
+
+    ptr = &D_80181230[arg0 * 8];
+    arg1 -= *ptr++;
+    if (arg1 < 0) {
+        return 0;
+    }
+    *arg3++ = arg1;
+
+    temp = *ptr++ - arg1;
+    if (temp <= 0) {
+        return 0;
+    }
+    *arg3++ = temp;
+
+    temp = D_80182168[*ptr++];
+    if (temp) {
+        temp = arg1 / temp;
+    } else {
+        temp = 0;
+    }
+
+    temp = *ptr++ - temp;
+    if ((temp <= arg2) && (arg2 < *ptr++)) {
+        return (arg2 - 0x8000) - temp;
+    }
+    return 0;
+}
+
+// ID 0x2C
+void EntityAlucardWaterEffect(Entity* arg0) {
+    s16 sp10[2];
+    s16 sp18;
+    s16 sp20;
+    u16 sp28;
+    Entity* var_s0;
+    s16 var_s1;
+    s16 var_s3;
+    s16 i;
+    u16 sp30;
+    u16 var_s6;
+    u32 var_s7;
+    u16 sp4A;
+    s32 gPlayerUnk0C;
+    Tilemap* tilemap = &g_Tilemap;
+    Entity* player = &PLAYER;
+
+    sp18 = player->posX.i.hi + tilemap->scrollX.i.hi;
+    gPlayerUnk0C = g_Player.unk0C;
```

Comment:
you can rename `gPlayerUnk0C` as `status` and use the various `PLAYER_STATUS_` to document the bitfields

Reply:
Awesome idea!